### PR TITLE
Validate conditions rule

### PR DESCRIPTION
**Pull request by aireilly**
Adds a Vale rule for ifdef conditions checking. 

Also added a ValidConditions.tengo file to play with tengo scripting.

### CODE REVIEW

**Comment by aireilly** [comment]
Would be nice to report errors on extra `endif::[]` statments. Doesn't currently.

**Comment by rohennes** [comment]
LGTM @aireilly , you happy to merge?

**Comment by aireilly** [comment]
Not quite ready yet - Need to account for rogue endif:[] statements. Also yes let's leave the tengo file in place. That'll be the model for developing future rules.

File: .vale/styles/AsciiDoc/ValidConditions.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  if_regex := "(ifdef::[0-9a-zA-Z_\\-]+\\[\\]|ifndef::[0-9a-zA-Z_\\-]+\\[\\]|ifeval::\\[.*\\])"
+  endif_regex := "endif::.*"
+
+  for line in text.split(scope, "\n") {
+    if text.re_match(if_regex, line) {
+      start := text.index(scope, line)
+      matches = append(matches, {begin: start, end: start + len(line)})
+    } else if text.re_match(endif_regex, line) {
+      if len(matches) > 0 {
```
Comment (aireilly) [inline .vale/styles/AsciiDoc/ValidConditions.yml:21]: To handle orphan endifs, maybe an additional clause `if len(matches) == 0` or something similar

**Comment by rohennes** [comment]
lgtm - merge?

**Comment by aireilly** [comment]
Yeah goferrit

**Pull request merged**